planner_cspace/src/planner_3d.cpp

```diff
@@ -241,24 +283,42 @@ class Planner3dNode
     grid_metric_converter::metric2Grid(
         map_info_, s[0], s[1], s[2],
         req.start.pose.position.x, req.start.pose.position.y, tf2::getYaw(req.start.pose.orientation));
-    s[2] = 0;
     grid_metric_converter::metric2Grid(
         map_info_, e[0], e[1], e[2],
         req.goal.pose.position.x, req.goal.pose.position.y, tf2::getYaw(req.goal.pose.orientation));
-    e[2] = 0;
+    ROS_INFO("Path plan from (%d, %d) to (%d, %d)", s[0], s[1], e[0], e[1]);
 
-    if (!(cm_rough_.validate(s, range_) && cm_rough_.validate(e, range_)))
+    const int tolerance_range = std::lround(req.tolerance / map_info_.linear_resolution);
+    const DiscretePoseStatus start_status = relocateDiscretePoseIfNeeded(s, tolerance_range, tolerance_angle_, true);
+    const DiscretePoseStatus goal_status = relocateDiscretePoseIfNeeded(e, tolerance_range, tolerance_angle_, true);
+    if (start_status == DiscretePoseStatus::OUT_OF_MAP)
     {
-      ROS_ERROR("Given start or goal is not on the map.");
+      ROS_ERROR("Given start is not on the map.");
       return false;
     }
-    else if (cm_rough_[s] == 100 || cm_rough_[e] == 100)
+    else if (start_status == DiscretePoseStatus::IN_ROCK)
     {
-      ROS_ERROR(
-          "Given start or goal is in Rock. (start: %d, end: %d)",
-          cm_rough_[s], cm_rough_[e]);
+      ROS_ERROR("Given start is in Rock.");
+      return false;
+    }
+    else if (start_status == DiscretePoseStatus::RELOCATED)
+    {
+      ROS_INFO("Given start is moved (%d, %d)", s[0], s[1]);
+    }
+    if (goal_status == DiscretePoseStatus::OUT_OF_MAP)
+    {
+      ROS_ERROR("Given goal is not on the map.");
       return false;
     }
+    else if (goal_status == DiscretePoseStatus::IN_ROCK)
+    {
+      ROS_ERROR("Given goal is in Rock.");
+      return false;
+    }
+    else if (goal_status == DiscretePoseStatus::RELOCATED)
+    {
+      ROS_INFO("Given goal is moved (%d, %d)", e[0], e[1]);
+    }
```

Comment:
If-chains are better to be changed to switch-case.
e.g.
```suggestion
    switch (goal_status)
    {
    case DiscretePoseStatus::OUT_OF_MAP:
      ROS_ERROR("Given goal is not on the map.");
      return false;
    case DiscretePoseStatus::IN_ROCK:
      ROS_ERROR("Given goal is in Rock.");
      return false;
    case DiscretePoseStatus::RELOCATED:
      ROS_INFO("Given goal is moved (%d, %d)", e[0], e[1]);
      break;
    }
```

Reply:
Done